animations/shared-element-transitions/index.md

```diff
@@ -0,0 +1,684 @@
+What happens when the InkWell receives a tap gesture?
+The following describes how Flutter performs the transition.
+
+<img src="images/hero-transition-0.png" alt="Before the transition the source hero appears in the source route">
+
+Before transition, the source Hero waits in the source route's widget
+tree. The destination route does not yet exist, and the overlay
+is blank and disabled.
+
+---
+
+<img src="images/hero-transition-1.png" alt="Transition begins and Destination Hero appears in the overlay">
+
+Tapping the Hero's InkWell triggers the animation, and the
+following occurs:
+
+* Offscreen, Flutter calculates the transition (including
+  position and size changes) from beginning to end,
+  moving the Hero along a curved path as described in the
+  Material motion spec. Flutter saves the size and location
+  of the Hero at the beginning and end of the transition.
+
+* Flutter places the destination Hero in the overlay, at the
+  same location and size as the _source_ Hero.
+
+* Flutter enables the overlay, obscuring the current route.
+
+* Flutter creates the destination route, placing it at the
+  top of the Navigator's stack (above the source route),
+  but it's obscured by the overlay.
+
+* The destination route won't contain the destination Hero
+  until the transition is complete.
+
+---
+
+<img src="images/hero-transition-2.png" alt="The Hero flies in the overlay to its final position and size">
+
+While the Hero is in-flight:
+
+* The same Animation object drives both
+  the route transition and the Hero animation.
+
+* As the Hero flies, its rectangular bounds are animated using
+  [Tween&lt;Rect&gt;,](https://docs.flutter.io/flutter/animation/Tween-class.html)
+  accessible via the Hero's `createRectTween` property.
+  By default, Flutter uses an instance of
+  [MaterialRectArcTween,](https://docs.flutter.io/flutter/material/MaterialRectArcTween-class.html)
+  which animates the rectangle's opposing corners along a curved path.
+  (See the [radial transformations](#radial-transformations) section
+  for an example that uses a different Tween animation.)
+
+---
+
+<img src="images/hero-transition-3.png" alt="When the transition is complete, the Hero is moved from the overlay to the destination route">
+
+When the flight completes:
+
+* Flutter moves the Hero widget from the overlay to the
+  destination route.
+
+* Flutter disables the (now blank) overlay.
+
+* The destination Hero appears in its final position in
+  the destination route.
+
+---
+
+Popping the route performs the same process, animating the
+Hero back to its size and location in the source route.
+
+### Essential classes
+
+The examples in this guide use the following classes to
+implement shared element transitions:
+
+[Hero](https://docs.flutter.io/flutter/widgets/Hero-class.html)
+: The "shared element" that flies from the source to the destination
+  route. The Hero itself doesn't fly, it's actually the image that
```

Comment:
This isn't right, it implies that we're flying a snapshot of the Hero or something.

What really happens is that we animate the Hero widget's bounds. That's really all there is to that.